api/src/main/java/io/minio/MinioProperties.java

```diff
@@ -35,42 +35,39 @@
 
   public String getVersion() {
     String result = version.get();
-    if (result == null) {
-      synchronized (INSTANCE) {
-        if (version.get() == null) {
-          try {
-            ClassLoader classLoader = getClass().getClassLoader();
-            setMinioClientJavaVersion(classLoader);
-            setDevelopmentVersion();
-          } catch (IOException e) {
-            LOGGER.log(Level.SEVERE, "IOException occured", e);
-            version.set("unknown");
-          }
-          result = version.get();
-        }
-      }
+    if (result != null) {
+      return result;
     }
-    return result;
+    setVersion();
+    return version.get();
   }
 
-  private void setDevelopmentVersion() {
-    if (version.get() == null) {
-      version.set("dev");
+  private synchronized void setVersion() {
+    if (version.get() != null) {
+      return;
+    }
+    version.set("dev");
+    ClassLoader classLoader = getClass().getClassLoader();
+    if (classLoader == null) {
+      return;
     }
-  }
 
-  private void setMinioClientJavaVersion(ClassLoader classLoader) throws IOException {
-    if (classLoader != null) {
+    try {
       Enumeration<URL> resources = classLoader.getResources("META-INF/MANIFEST.MF");
       while (resources.hasMoreElements()) {
-        Manifest manifest = new Manifest(resources.nextElement().openStream());
-        for (Object k : manifest.getMainAttributes().keySet()) {
-          String versionString = "Implementation-Version";
-          if (k.toString().equals(versionString)) {
-            version.set(manifest.getMainAttributes().getValue((Attributes.Name) k));
+        try (InputStream is = resources.nextElement().openStream()) {
+          Manifest manifest = new Manifest(is);
+          String implementationTitleValue =
+              manifest.getMainAttributes().getValue("Implementation-Title");
+          if (implementationTitleValue != null && implementationTitleValue.equals("minio")) {
```

Comment:
Just reduce to `if ("minio".equals(manifest.getMainAttributes().getValue("Implementation-Title")))`

Reply:
👍  fixed